pkg/puppetdb/nodes.go

```diff
@@ -2,15 +2,35 @@ package puppetdb
 
 import (
 	"fmt"
+	"strconv"
 )
 
+func (p Pagination) toParams() map[string]string {
+	params := map[string]string{}
+	if p.Limit > 0 {
+		params["limit"] = strconv.Itoa(p.Limit)
+	}
+	if p.Offset > 0 {
+		params["offset"] = strconv.Itoa(p.Offset)
+	}
+
+	if p.IncludeTotal {
+		params["include_total"] = strconv.FormatBool(p.IncludeTotal)
+	}
+	return params
+}
+
 // Nodes will return all nodes matching the given query. Deactivated and expired nodes aren’t included in the response.
-func (c *Client) Nodes(query string) ([]Node, error) {
+func (c *Client) Nodes(query string, pagination Pagination) ([]Node, error) {
 	payload := []Node{}
 	req := c.resty.R().SetResult(&payload)
 	if query != "" {
 		req.SetQueryParam("query", query)
 	}
+	if pagination.toParams() != nil {
```

Comment:
I think you can omit this, your calling toParams() twice, which will create two maps.  
I because you initialize the params map on line 8, it will return an empty map if no options are set, so it should be safe to just write ```req.SetQueryParams(pagination.toParams())```